control_panel_api/tests/test_views.py

```diff
@@ -156,6 +162,14 @@ def test_create_when_name_invalid(self):
         response = self.client.post(reverse('s3bucket-list'), data)
         self.assertEqual(HTTP_400_BAD_REQUEST, response.status_code)
 
+    @patch('control_panel_api.services.create_bucket')
+    @patch('control_panel_api.services.create_bucket_policies')
+    def test_create_calls_apis(self, mock_create_bucket, mock_create_bucket_policies):
+        data = {'name': 'test-bucket-123'}
+        self.client.post(reverse('s3bucket-list'), data)
+        mock_create_bucket.assert_called()
+        mock_create_bucket_policies.assert_called()
+
     def test_update_when_valid_data(self):
         data = {'name': 'test-bucket-updated'}
```

Comment:
I don't know how to enforce this in django but we probably want to prevent S3 buckets renaming as it's non-trivial and we should treat S3 buckets are permanent things as apps, IAM policies, etc...refer to them so it's easy to break things if renaming them.

Also see: https://stackoverflow.com/questions/41511034/how-to-rename-aws-s3-bucket-name

Reply:
Yep no renames for the timebeing